pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -986,27 +966,20 @@ func (p *PriorityQueue) Update(logger klog.Logger, oldPod, newPod *v1.Pod) error
 	if usPodInfo := p.unschedulablePods.get(newPod); usPodInfo != nil {
 		pInfo := updatePod(usPodInfo, newPod)
 		p.updateNominatedPodUnlocked(logger, oldPod, pInfo.PodInfo)
-		if isPodUpdated(oldPod, newPod) {
-			gated := usPodInfo.Gated
-			if p.isPodBackingoff(usPodInfo) {
-				if err := p.podBackoffQ.Add(pInfo); err != nil {
-					return err
-				}
-				p.unschedulablePods.delete(usPodInfo.Pod, gated)
-				logger.V(5).Info("Pod moved to an internal scheduling queue", "pod", klog.KObj(pInfo.Pod), "event", PodUpdate, "queue", backoffQ)
-			} else {
-				if added, err := p.addToActiveQ(logger, pInfo); !added {
-					return err
-				}
-				p.unschedulablePods.delete(usPodInfo.Pod, gated)
-				logger.V(5).Info("Pod moved to an internal scheduling queue", "pod", klog.KObj(pInfo.Pod), "event", BackoffComplete, "queue", activeQ)
-				p.cond.Broadcast()
-			}
-		} else {
-			// Pod update didn't make it schedulable, keep it in the unschedulable queue.
-			p.unschedulablePods.addOrUpdate(pInfo)
+		gated := usPodInfo.Gated
+		// When unschedulable Pods are updated, we check with QueueingHint
+		// whether the update may make the pods schedulable.
+		// Plugins have to implement a QueueingHint for Pod/Update event
+		// if the rejection from them could be resolved by updating unschedulable Pods itself.
+		hint := p.isPodWorthRequeuing(logger, pInfo, UnschedulablePodUpdate, oldPod, newPod)
+		queue := p.requeuePodViaQueueingHint(logger, pInfo, hint, UnschedulablePodUpdate.Label)
+		logger.V(5).Info("Pod moved to an internal scheduling queue", "pod", klog.KObj(newPod), "event", PodUpdate, "queue", queue)
```

Comment:
```suggestion
		logger.V(5).Info("Pod moved to an internal scheduling queue because the Pod is updated", "pod", klog.KObj(newPod), "event", PodUpdate, "queue", queue)
```

Reply:
Also, we should log this only when pod is moved to activeQ/backoffQ, shouldn't we?